src/lib/datepicker/datepicker-input.ts

```diff
@@ -156,17 +156,19 @@ export class MatDatepickerInput<D> implements AfterContentInit, ControlValueAcce
   get disabled(): boolean { return !!this._disabled; }
   set disabled(value: boolean) {
     const newValue = coerceBooleanProperty(value);
+    const element = this._elementRef.nativeElement;
 
     if (this._disabled !== newValue) {
       this._disabled = newValue;
       this._disabledChange.emit(newValue);
     }
 
-    if (newValue) {
+    // We need to null check the `blur` method, because it's undefined during SSR.
+    if (newValue && typeof element.blur === 'function') {
```

Comment:
can we just do `newValue && element.blur`?

Reply:
We could, I was trying to avoid some super edge case where they might decide to define blur as an object which isn't a function.

Reply:
I may be biased from following the Google style guide for a while. But their philosophy is to go with the simpler check and just assume people aren't going to do insane things like replacing methods on native elements. I'll leave it up to you though.